packages/medusa/src/services/discount.ts

```diff
@@ -264,7 +269,9 @@ class DiscountService extends BaseService {
 
         if (conditions?.length) {
           for (const cond of conditions) {
```

Comment:
Promise.all here in order to take advantage of the I/O, when the computation are sent to the event loop the I/O operations are handled by sub system and can be therefore parallel unlike the computational operations